### PR TITLE
Add .net9 transitive dependencies

### PR DESCRIPTION
**Pull request by brettsam**
### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 1]
```diff
@@ -23,16 +23,18 @@
...
-    <PackageReference Include="System.Text.Encodings.Web" Version="8.0.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="8.0.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="8.0.1" />
   </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-rc.2.24473.5" />
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:37]: Yes. As soon as those are out, we can update and merge.

[the remark this answers is not in the document]
Comment (brettsam) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:37]: yep

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 2]
```diff
@@ -23,16 +23,18 @@
...
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.1" />
   </ItemGroup>
-  
+
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="8.0.2" />
-    <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
-    <PackageReference Include="Microsoft.Extensions.Options" Version="8.0.2" />
-    <PackageReference Include="System.Text.Encodings.Web" Version="8.0.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="8.0.0" />
```
Comment (kshyju) [inline src/DotNetWorker.Core/DotNetWorker.Core.csproj:29]: System.Collections.Immutable has a 9X version. Shouldn't we use that for net9 TFM case?

Comment (jviau) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:29]: We will be - it is a transitive reference of `Microsoft.Extensions.Hosting`, which is 9.0.0 with `net9` tfm

File: src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -54,7 +54,7 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
             services.TryAddSingleton<IWorker, GrpcWorker>();
             services.TryAddSingleton<IInvocationHandler, InvocationHandler>();
 
-#if NET8_0_OR_GREATER
+#if NET9_0_OR_GREATER
```
Comment (kshyju) [inline src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs:57]: Do we need this change?

Comment (brettsam) [reply src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs]: put it back to net6

**Review by liliankasem — APPROVED** [review]

File: src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs (hunk 3 above)
Comment (brettsam) [reply src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs]: fixed

**Review by fabiocav — APPROVED** [review]

**Review by kshyju — APPROVED** [review]

**Review by jviau — APPROVED** [review]

**Review by fabiocav — APPROVED** [review]

**Pull request merged**